handsontable/src/plugins/dialog/dialog.js

```diff
@@ -0,0 +1,364 @@
+    return !!this.hot.getSettings()[PLUGIN_KEY];
+  }
+
+  /**
+   * Enable plugin for this Handsontable instance.
+   */
+  enablePlugin() {
+    if (this.enabled) {
+      return;
+    }
+
+    this.#ui = new DialogUI({
+      rootElement: this.hot.rootWrapperElement,
+    });
+    this.registerShortcuts();
+
+    super.enablePlugin();
+  }
+
+  /**
+   * Update plugin state after Handsontable settings update.
+   */
+  updatePlugin() {
+    this.disablePlugin();
+    this.enablePlugin();
+
+    super.updatePlugin();
+  }
+
+  /**
+   * Disable plugin for this Handsontable instance.
+   */
+  disablePlugin() {
+    this.#destroyDialog();
+    this.unregisterShortcuts();
+
+    super.disablePlugin();
+  }
+
+  /**
+   * Check if the dialog is currently visible.
+   *
+   * @returns {boolean} True if the dialog is visible, false otherwise.
+   */
+  isVisible() {
+    return this.#isVisible;
+  }
+
+  /**
+   * Register shortcuts responsible for toggling a merge.
+   *
+   * @private
+   */
+  registerShortcuts() {
+    const manager = this.hot.getShortcutManager();
+    const pluginContext = manager.getContext(SHORTCUTS_CONTEXT_NAME) ||
+      manager.addContext(SHORTCUTS_CONTEXT_NAME, 'global');
+
+    pluginContext.addShortcut({
+      keys: [['Escape']],
+      callback: () => {
+        this.hide();
+      },
+      runOnlyIf: () => this.#isVisible && this.getSetting('closable'),
+      group: SHORTCUTS_GROUP,
+    });
+  }
+
+  /**
+   * Unregister shortcuts responsible for toggling a merge.
+   *
+   * @private
+   */
+  unregisterShortcuts() {
+    const shortcutManager = this.hot.getShortcutManager();
+    const pluginContext = shortcutManager.getContext(SHORTCUTS_CONTEXT_NAME);
+
+    pluginContext.removeShortcutsByGroup(SHORTCUTS_GROUP);
+  }
+
+  /**
+   * Show dialog with given configuration.
+   * Displays the dialog with the specified content and settings.
+   *
+   * @param {object} settings Dialog configuration object containing content and display options.
+   */
+  show(settings = {}) {
+    if (!this.enabled || this.isVisible()) {
+      return;
+    }
+
+    this.hot.runHooks('beforeDialogShow');
+
+    this.hot.getShortcutManager().setActiveContextName(SHORTCUTS_CONTEXT_NAME);
+    this.update(settings);
+    this.#ui.showDialog(this.getSetting('animation'));
+    this.#isVisible = true;
+
+    this.hot.runHooks('afterDialogShow');
+  }
+
+  /**
+   * Hide the currently open dialog.
+   * Closes the dialog and cleans up event listeners.
+   */
+  hide() {
+    if (!this.isVisible()) {
+      return;
+    }
+
+    this.hot.runHooks('beforeDialogHide');
+
+    this.#ui.hideDialog(this.getSetting('animation'));
+    this.hot.getShortcutManager().setActiveContextName('grid');
+    this.#isVisible = false;
+
+    this.hot.runHooks('afterDialogHide');
+  }
+
+  /**
+   * Update the dialog configuration.
+   *
+   * @param {object} settings - The configuration to update the dialog with.
+   */
+  update(settings) {
```

Comment:
It's worth extending the docs for the `update` and `show` methods' options. Now it looks like this

<img width="657" height="723" alt="Image" src="https://github.com/user-attachments/assets/a923ef00-5b33-4cb3-becb-435a7bd45c7b" />